src/monattack.cpp

```diff
@@ -4217,10 +4296,81 @@ void mattack::suicide(monster *z, int index)
 {
     (void)index;
     Creature *target = z->attack_target();
-    if( target == nullptr || 
-        rl_dist( z->pos(), target->pos() ) > 2 || 
+    if( target == nullptr ||
+        rl_dist( z->pos(), target->pos() ) > 2 ||
         !z->sees( *target ) ) {
         return;
     }
     z->die(z);
 }
+
+void mattack::stretch_attack(monster *z, int index){
+    if( !z->can_act() ) {
+        return;
+    }
+    Creature *target = z->attack_target();
+    if (target == nullptr || rl_dist(z->pos(), target->pos()) > 3 || !z->sees(*target)){
```

Comment:
"Arms" should have no problem going through bars; chain fences, etc not so much.